test/unittest/rtps/builtin/BuiltinDataSerializationTests.cpp

```diff
@@ -2767,6 +2767,188 @@ TEST(BuiltinDataSerializationTests, optional_qos_extensions_writer)
     ASSERT_EQ(wpd.writer_resource_limits->reader_filters_allocation.increment, 6u);
 }
 
+/**
+ * This test checks that a correct serialization is obtained
+ * when non-default eProsima's optional qos extensions are used in ReaderProxyData.
+ * History Qos is additionally tested.
+ */
+TEST(BuiltinDataSerializationTests, optional_qos_extensions_reader_serialize_non_default_values)
```

Comment:
Apart from these unittests which look good to me. It would be nice if we could add a couple of BB tests that:

* Check that only non-default qos are sent 
* Property of optional qos at DDS level

We may use the test transport for that.